src/main/java/de/metas/ui/web/handlingunits/process/WEBUI_M_HU_Pick.java

```diff
@@ -125,9 +137,87 @@ public Object getParameterDefaultValue(final IProcessDefaultParameter parameter)
 			lookupSource = LookupSource.lookup)
 	private LookupValuesList getShipmentScheduleValues(final LookupDataSourceContext context)
 	{
-		return createNewDefaultParametersFiller().getShipmentScheduleValues(context);
+
+		final HUEditorRow huEditorRow = getSingleHUEditorRow();
+
+		// Get all the matching shipment schedules and then filter for schedules which are not picked and match the order-line attributes.
+		return createNewDefaultParametersFiller()
+				.getShipmentScheduleValues(context)
```

Comment:
this seems to be the only invocation of `WEBUI_M_HU_Pick_ParametersFiller.getShipmentScheduleValues(..)`, 
so why not have all this and the code it invokes in WEBUI_M_HU_Pick_ParametersFiller?

Reply:
There is no specific reason for adding all the code in this class. I wasn't sure that this is the only invocation of WEBUI_M_HU_Pick_ParametersFiller.getShipmentScheduleValues(..). I chose to add all the logic here, because I was thinking that the business logic related to the process should stay in this class.